include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,168 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    container(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table)
+    :
+    container(table),
+    linear_index(container->values.size())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const AccessorBase<T, false> &a)
+    :
+    container(a.container),
+    linear_index(a.linear_index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table,
+                                           const std::ptrdiff_t         index)
+    :
+    container(table),
+    linear_index(index)
+  {}
```

Comment:
I think you could check for validity of the index here, i.e., whether it is in range for the container.

Reply:
Yes; I will add this.